lib/src/config.rs

```diff
@@ -77,6 +79,7 @@ pub struct ConfigBuilder {
     db: Database,
     fetch_size: usize,
     max_connections: usize,
+    client_certificate_provider: Option<Box<dyn ClientCertificateProvider>>,
```

Comment:
Following the above comment, this one would be the certificate directly

```suggestion
    client_certificate: Option<ClientCertificate>,
```